### PR TITLE
Fix/close file

### LINKED ISSUE #35: FileChannel is not closed
`DownloadHandler` seems does not close `FileChannel` after downloading.

https://github.com/eed3si9n/gigahorse/blob/0.3.x/core/src/main/scala/gigahorse/DownloadHandler.scala#L29-L33

### PR DESCRIPTION
**Pull request by jkugiya**
close #35

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @jkugiya,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: core/src/main/scala/gigahorse/Stream.scala [hunk 1]
```diff
@@ -31,7 +31,11 @@ abstract class Stream[A] {
   def foreach(f: A => Unit): Future[Unit]
 
   /** Runs f on each element received to the stream with its previous output. */
-  def fold[B](zero: B)(f: (B, A) => B): Future[B]
+  final def fold[B](zero: B)(f: (B, A) => B): Future[B] =
+    foldResource(zero)(f, () => ())
```
Comment (eed3si9n) [inline core/src/main/scala/gigahorse/Stream.scala]: I think it makes `Stream` more useful if could keep this free of implementation.

**Review by eed3si9n — CHANGES_REQUESTED** [review]
Thanks for the contribution!

One comment.

File: core/src/main/scala/gigahorse/Stream.scala (hunk 1 above)
Comment (jkugiya) [reply core/src/main/scala/gigahorse/Stream.scala]: Thanks for the feedback. I reverted `Stream`'s fold implementation for flexibility, and moved it to concrete classes.

**Pull request merged**